### PR TITLE
Compute top-level spans on the tracer side

### PR DESCRIPTION
**Pull request by kevingosse**
This is a requirement to implement flushing of partial traces.

### CODE REVIEW

File: src/Datadog.Trace/Metrics.cs [hunk 1]
```diff
@@ -21,5 +21,11 @@ internal static class Metrics
         /// Read: Rate Limiter Priority Sample Rate
         /// </summary>
         public const string SamplingLimitDecision = "_dd.limit_psr";
+
+        /// <summary>
+        /// To be set on top-level spans
+        /// Top-level spans are spans with a different service name from their immediate parent
```
Comment (lucaspimentel) [inline src/Datadog.Trace/Metrics.cs:27]:
```suggestion
        /// Identifies top-level spans.
        /// Top-level spans have a different service name from their immediate parent or have no parent.
```

File: src/Datadog.Trace/AgentHttpHeaderNames.cs [hunk 2]
```diff
@@ -38,6 +38,11 @@ internal static class AgentHttpHeaderNames
         /// </summary>
         public const string ContainerId = "Datadog-Container-ID";
 
+        /// <summary>
+        /// Tells the agent whether top-level spans are computed by the tracer
+        /// </summary>
+        public const string ComputedTopLevel = "Datadog-Client-Computed-Top-Level";
```
Comment (lucaspimentel) [inline src/Datadog.Trace/AgentHttpHeaderNames.cs:44]: We can't change the header name, but would you be ok with renaming the property for clarity?
```suggestion
        public const string ComputedTopLevelSpan = "Datadog-Client-Computed-Top-Level";
```

**Review by lucaspimentel — APPROVED** [review]
LGTM. Just tweaking some comments and a property name.

**Comment by lucaspimentel** [comment]
Support for `_dd.top_level` was added in Agent 7.25 (https://github.com/DataDog/datadog-agent/pull/6941). Older versions will just ignore the new http header and the new metric, right?https://github.com/DataDog/datadog-agent/pull/6941

File: test/Datadog.Trace.Tests/SpanTests.cs [hunk 3]
```diff
@@ -135,5 +138,28 @@ public void Accurate_Duration(int minimumSleepMilliseconds)
...
+            var spans = new List<(Scope Scope, bool IsTopLevel)>();
+
+            spans.Add((_tracer.StartActive("Root", serviceName: "root"), true));
+            spans.Add((_tracer.StartActive("Child1", serviceName: "root"), false));
+            spans.Add((_tracer.StartActive("Child2", serviceName: "child"), true));
+            spans.Add((_tracer.StartActive("Child3", serviceName: "child"), false));
+            spans.Add((_tracer.StartActive("Child4", serviceName: "root"), true));
+
+            // Dispose the scope to populate the top-level property
+            foreach (var (scope, _) in spans.AsEnumerable().Reverse())
```
Comment (zacharycmontoya) [inline test/Datadog.Trace.Tests/SpanTests.cs:154]: You don't need to dispose these anymore to set the top-level property, do you?

**Review by zacharycmontoya — APPROVED** [review]

**Review by andrewlock — APPROVED** [review]

File: test/Datadog.Trace.Tests/SpanTests.cs (hunk 3 above)
Comment (lucaspimentel) [reply test/Datadog.Trace.Tests/SpanTests.cs:154]: I was going to ask the same question, but then thought it doesn't hurt (although the comment may be wrong). Thanks for asking 😅

File: src/Datadog.Trace/Metrics.cs (hunk 1 above)
Comment (kevingosse) [reply src/Datadog.Trace/Metrics.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/1303/commits/87ab80ddc9107b9b6cff3028a319b5824358d6e7

File: src/Datadog.Trace/AgentHttpHeaderNames.cs (hunk 2 above)
Comment (kevingosse) [reply src/Datadog.Trace/AgentHttpHeaderNames.cs]: Done in https://github.com/DataDog/dd-trace-dotnet/pull/1303/commits/87ab80ddc9107b9b6cff3028a319b5824358d6e7

File: test/Datadog.Trace.Tests/SpanTests.cs (hunk 3 above)
Comment (kevingosse) [reply test/Datadog.Trace.Tests/SpanTests.cs]: You're right. Removed in https://github.com/DataDog/dd-trace-dotnet/pull/1303/commits/87ab80ddc9107b9b6cff3028a319b5824358d6e7

**Pull request merged**